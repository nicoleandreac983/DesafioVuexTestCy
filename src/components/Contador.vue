<template>
  <h2 class="titulo-principal">{{ tituloPorDefecto }}</h2>
  <p class="counter">{{ contador }}</p>

  <div>
    <button class="btn primary incrementar" v-on:click="incrementar">+1</button>
    <button class="btn warning disminuir" v-on:click="disminuir">-1</button>
  </div>
</template>

<script>
export default {
  name: 'Contador',
  props: {
    titulo: String,
    inicio: {
      default: 0, // Define que que el contador empiece en 0
    },
  },

  data() {
    return {
      contador: this.inicio,
    }
  },
  methods: {
    aumentar() {
      return this.contador + this.contador
    },

    incrementar() {
      this.contador++
    },

    disminuir() {
      this.contador--
    },
  },
  computed: {
    // Las propieades computadas se guardan en cache
    aumentarContador() {
      return this.contador * this.contador
    },

    tituloPorDefecto() {
      return this.titulo || 'Contador'
    },
  },
}
</script>

<style>
.btn {
  border-radius: 1rem;
  margin: 0 5px;
  width: 100px;
  padding: 5px;
  font-size: 1rem;
}
.primary {
  background-color: rgb(77, 77, 231);
  color: rgb(255, 255, 255);
}
.counter {
  font-size: 36px;
}

.warning {
  background-color: rgb(230, 53, 53);
  color: rgb(255, 255, 255);
}

.second {
  background-color: rgb(255, 165, 0);
  color: rgb(0, 0, 0);
}
</style>
