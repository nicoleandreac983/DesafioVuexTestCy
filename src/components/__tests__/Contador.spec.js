import { shallowMount } from '@vue/test-utils'
import Contador from '@/components/Contador.vue'
import { describe, it, expect, test } from 'vitest'

describe('Componente Contador.vue', () => {
  test('Validación de match con el snapshot', () => {
    //Selección el Sujeto de pruebas
    const wrapper = shallowMount(Contador)
    //Aserción
    expect(wrapper.html()).toMatchSnapshot()
  })
})

// funcionalidad de incremento y decremento del contador

describe('Componente Contador.vue', () => {
  test('Incrementar en 1 el valor del contador', async () => {
    const wrapper = shallowMount(Contador)
    const botonIncrementar = wrapper.find('button')
    botonIncrementar.trigger('click')
    const valor = wrapper.find('.incrementar').text()
    expect(valor).toBe('+1')
  })
})

describe('Componente Contador.vue', () => {
  test('Disminuir en 1 el valor del contador', async () => {
    const wrapper = shallowMount(Contador)
    const botonDisminuir = wrapper.find('button')
    botonDisminuir.trigger('click')
    const valor = wrapper.find('.disminuir').text()
    expect(valor).toBe('-1')
  })
})

// Verificar que se reciba un valor inicial en el contador

describe('Componente Contador.vue', () => {
  test('Valor inicial del contador es 0', () => {
    const wrapper = shallowMount(Contador)
    const valor = wrapper.find('.counter').text()
    expect(valor).toBe('0')
  })
})
