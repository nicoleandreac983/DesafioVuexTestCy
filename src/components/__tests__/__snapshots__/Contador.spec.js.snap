// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Contador.vue > Validaci√≥n de match con el snapshot 1`] = `
"<h2 class="titulo-principal">Contador</h2>
<p class="counter">0</p>
<div><button class="btn primary incrementar">+1</button><button class="btn warning disminuir">-1</button></div>"
`;
